### PR TITLE
[onert] Introduce tests for Reshape, Squeeze and ExpandDims

### PR DESCRIPTION
**Pull request by mbencer**
This commit adds more tests for Reshape, Squeeze and ExpandDims. The purpose is to cover the cases where memory copying of those ops can be skipped.

ONE-DCO-1.0-Signed-off-by: Mateusz Bencer m.bencer@partner.samsung.com

Draft: https://github.com/Samsung/ONE/pull/14057

### CODE REVIEW

File: tests/nnfw_api/src/GenModelTests/MemorySharingModels.test.cc [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+  CircleGen cgen;
+  std::vector<int32_t> new_shape_data{2, 2};
+  uint32_t new_shape_buf = cgen.addBuffer(new_shape_data);
+  int input = cgen.addTensor({{4}, circle::TensorType::TensorType_FLOAT32});
+  int cos1_out = cgen.addTensor({{4}, circle::TensorType::TensorType_FLOAT32});
+  int new_shape = cgen.addTensor({{2}, circle::TensorType::TensorType_INT32, new_shape_buf});
+  int reshape_out = cgen.addTensor({{2, 2}, circle::TensorType::TensorType_FLOAT32});
+  int cos2_out = cgen.addTensor({{2, 2}, circle::TensorType::TensorType_FLOAT32});
+
+  cgen.addOperatorCos({{input}, {cos1_out}});
```
Comment (zetwhite) [inline tests/nnfw_api/src/GenModelTests/MemorySharingModels.test.cc:34]: Just for curiosity,

Is there a specific reason to choose `cos` operator? 
Did you choose `cos` because it is element-wise operator - which works without additional arguments?

Comment (mbencer) [reply tests/nnfw_api/src/GenModelTests/MemorySharingModels.test.cc:34]: This cos node is needed just to not set Reshape input/output as a input/output of the whole model (in such a case sharing memory is disabled). But I see your confusion so I've wrapped it in helper function.

Comment (mbencer) [reply tests/nnfw_api/src/GenModelTests/MemorySharingModels.test.cc]: > But I see your confusion so I've wrapped it in helper function.

Sorry, I've forgot to push the code last time ;)

**Review by zetwhite — DISMISSED** [review]

**Comment by hseok-oh** [comment]
@mbencer Please add negative test: same or more negative test with positive test.

**Comment by mbencer** [comment]
FYI: I will be on vacation until 26 (Nov) so please expect delays in communication ;-)

**Comment by mbencer** [comment]
> > @mbencer Please add negative test: same or more negative test with positive test to fulfill release criteria.
> 
> I see the argument about release criteria. Due to fact that compilation/inference should run without errors both with and without my optimization it's hard to write such negative unit test explicitly.
> 
> To fulfill the requirements I've just added some general negative tests for the related operators. Now only 5 (at least 12 is expected). If you are ok with such approach I'll try to add more.

@hseok-oh Please let me know if you are ok with such approach ;)

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-x64-release

File: tests/nnfw_api/src/GenModelTests/one_op_tests/ExpandDims.test.cc [hunk 2]
```diff
@@ -75,6 +75,25 @@ TEST_F(GenModelTest, OneOp_neg_ExpandDims_Axis)
   SUCCEED();
 }
 
+TEST_F(GenModelTest, OneOp_neg_ExpandDims_NegAxis)
```
Comment (hseok-oh) [inline tests/nnfw_api/src/GenModelTests/one_op_tests/ExpandDims.test.cc:78]: Negative test's name should start with `neg_`. Please remove `OneOp_` prefix. (Same on other negative tests)

**Review by hseok-oh — APPROVED** [review]
LGTM

**Review by ragmani — APPROVED** [review]
LGTM

**Comment by hseok-oh** [comment]
@zetwhite @glistening ping

**Review by glistening — APPROVED** [review]
LGTM
